src/utils/helpers/source.helper.ts

```diff
@@ -224,47 +229,57 @@ export class SourceHelper {
    *
    * @param id The ID of the player
    */
-  static hook(id: string): void {
-    const elementId = `#player-${id}`;
-
-    DOMHelper
-      .watch(elementId)
-      .then(e => e[0] as HTMLVideoElement)
-      .then(player => {
-        const store = useSourcesStore();
+  static async hook(id: string): Promise<void> {
```

Comment:
The `hook` method could be the culprit here. We shouldn't loop sources unnecessarily.